chain/chain/src/flat_storage_resharder.rs

```diff
@@ -393,12 +393,9 @@ fn shard_split_handle_key_value(
         col::DELAYED_RECEIPT_OR_INDICES
         | col::PROMISE_YIELD_INDICES
         | col::PROMISE_YIELD_TIMEOUT
-        | col::PROMISE_YIELD_RECEIPT
-        | col::BUFFERED_RECEIPT_INDICES
-        | col::BUFFERED_RECEIPT => {
-            // TODO(trisfald): implement logic and remove error log
-            let col_name = ALL_COLUMNS_WITH_NAMES[key_column_prefix as usize].1;
-            error!(target: "resharding", "flat storage resharding of {col_name} is not implemented yet!");
+        | col::PROMISE_YIELD_RECEIPT => copy_kv_to_all_children(&status, key, value, store_update),
+        col::BUFFERED_RECEIPT_INDICES | col::BUFFERED_RECEIPT => {
+            copy_kv_to_left_child(&status, key, value, store_update)
```

Comment:
In the future, could be nice to put the function to some place where I could reuse it in `retain_split_shard`: https://github.com/near/nearcore/blob/master/core/store/src/trie/mem/resharding.rs#L35

For example, it could return `RetainQuery` enum which could be `All`, `LeftOnly` or `SplitByAccount`.